bin/run-tests.sh

```diff
@@ -29,9 +29,8 @@ for test_dir in tests/*; do
     done
 
     echo "${test_dir_name}: comparing results.json to expected_results.json"
-    diff "${results_file_path}.tmp" "${expected_results_file_path}.tmp"
 
-    if [ $? -ne 0 ]; then
+    if ! diff "${results_file_path}.tmp" "${expected_results_file_path}.tmp"; then
```

Comment:
I wouldn't mind a PR to the generic test runner for this changed. Easier to read